test/metric-end-results/YAMLMetrics.test.ts

```diff
@@ -0,0 +1,29 @@
+import { expectFileMetric, parseAllFileMetrics } from "./TestHelper";
+import { FileMetric, FileMetricResults } from "../../src/parser/metrics/Metric";
+
+describe("YAML metrics tests", () => {
+    const yamlTestResourcesPath = "./resources/yaml/";
+
+    let results: Map<string, FileMetricResults>;
+
+    const testFileMetric = (inputPath, metric, expected) =>
+        expectFileMetric(results, inputPath, metric, expected);
+
+    beforeAll(async () => {
+        results = await parseAllFileMetrics(yamlTestResourcesPath);
+    });
+
+    describe("parses YAML nesting level metric", () => {
+        it("should calculate nesting level", () => {
+            testFileMetric(yamlTestResourcesPath + "example.yaml", FileMetric.maxNestingLevel, 3);
+        });
+
+        it("should correctly calculate nesting level of a more complex yaml", () => {
+            testFileMetric(yamlTestResourcesPath + "tests.yml", FileMetric.maxNestingLevel, 6);
```

Comment:
I do not really get why this is 6 and not 5. I will have to think about it.

Reply:
Ok, seems to be because of this `|` multiline thingy. I guess its fine to count that as an additional nesting level (?)